### DISCUSSION TITLE
quarto line wrap

### DISCUSSION THREAD

**Discussion by kbvernon** [posit-dev/positron#4429] [Q&A] [answered]
I am looking for guidance on how to wrap text in a qmd file. I see that there are open issues re: line wrapping in the console, but I don't see any on line wrapping in the text editor, and I didn't see anything in discussions, either. I am aware of the rewrap extension, which should work on md files, but evidently not qmd files, or I can't seem to get it to work anyway. I also noticed that switching between Visual and Source editor does not enforce line wrapping like it does in RStudio. In fact, line wrapping was removed after switching to Visual then back to Source.

Am I doing something wrong, or is this a "wait for it to get implemented" scenario?

Thanks!

**Comment by juliasilge** [comment, marked as answer]
Have you checked out the line wrapping editor support in Quarto? You can specify how you want to wrap lines: <https://quarto.org/docs/visual-editor/vscode/#line-wrapping>

For functionality like what rewrap provides, I've opened this issue on the Quarto repo to see what we can do: https://github.com/quarto-dev/quarto/issues/408

Can you share a bit more detail about what you are observing when you switch between visual and source editors? I am not quite understanding.

**Reply by kbvernon** [reply]
Ah, it's an #RTFM situation. Sorry about that. I thought I had thoroughly searched the docs but evidently not.  

I can confirm that adding: 
```md
"quarto.visualEditor.markdownWrap": "column",
"quarto.visualEditor.markdownWrapColumn": 80
```
to `settings.json` works. Now, switching to Visual editor and then back to Source reformats all text to wrap at the specified character length, in this case, 80 characters. For example, if I type this in Source:

```md
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
```
then switch to Visual and back to Source, it now looks like this:
```md
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
incididunt ut labore et dolore magna aliqua. Ut enim
```
Anyway, thanks for the help on this! And sorry again for not reading the docs carefully!
